src/construction.cpp

```diff
@@ -1377,13 +1377,13 @@ void check_constructions()
             debugmsg("Unknown skill %s in %s", c->skill.c_str(), display_name.c_str());
         }
         c->requirements.check_consistency(display_name);
-        if (!c->pre_terrain.empty() && !c->pre_is_furniture && termap.count(c->pre_terrain) == 0) {
+        if (!c->pre_terrain.empty() && !c->pre_is_furniture && !ter_str_id( c->pre_terrain ).is_valid() ) {
```

Comment:
> Good point. How to treat invalid identifiers apart from reporting? Leave them as they are, or replace with  NULL_ID s?

You can't really do any meaningful. The program does not know whether a null-id is a suitable replacement (maybe the value is required to be a valid non-null id).

If you have an instance of one of the Json classes, use their error reporting functions (e.g. `JsonObject::throw_error`). Otherwise just issue a `debugmsg` and continue.